index.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    arr = to_a
+    return arr.to_enum unless block_given?
+
+    n = 0
+    while arr.length > n
+      yield(arr[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    arr = to_a
+    return arr.to_enum unless block_given?
+
+    n = 0
+    while arr.length > n
+      yield(arr[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    res = []
+    my_each do |v|
+      res << v if yield(v) == true
+    end
+    res
+  end
+
+  def my_all?(*args)
+    check = true
+    if block_given?
+      if args.empty?
+        my_each { |i| check = false unless yield(i) }
+      elsif args[0].is_a?(Class)
+        my_each { |i| check = false unless yield(i).class == args[0] }
+      elsif args[0].is_a?(Regexp)
+        my_each { |i| check = false unless (args[0]).match(yield(i)) }
+      else
+        my_each { |i| check = false unless yield(i) == args[0] }
+      end
+      return check
+    elsif args.empty?
+      my_all? { |obj| obj }
+    else
+      my_all?(args[0]) { |obj| obj }
+    end
+  end
+
+  def my_any?(*args)
+    check = false
+    if block_given?
+      if args.empty?
+        my_each { |i| check = true if yield(i) }
+      elsif args[0].is_a(Class)
+        my_each { |i| check = true if yield(i).class == args[0] }
+      elsif args[0].is_a(Regexp)
+        my_each { |i| check = true if yield(i).match(args[0]) }
+      else
+        my_each { |i| check = true if yield(i) == args[0] }
+      end
+      return check
+    elsif args.empty?
+      my_any? { |obj| obj }
+    else
+      my_any?(args[0]) { |obj| obj }
+    end
+  end
+
+  def my_none?(*args)
+    check = true
+    if block_given?
+      if args.empty?
+        my_each { |i| check = false if yield(i) }
+      elsif args[0].is_a(Class)
+        my_each { |i| check = false if yield(i).class == args[0] }
+      elsif args[0].is_a(Regexp)
+        my_each { |i| check = false if yield(i).match(args[0]) }
+      else
+        my_each { |i| check = false if yield(i) == args[0] }
+      end
+      return check
+    elsif args.empty?
+      my_none? { |obj| obj }
+    else
+      my_none?(args[0]) { |obj| obj }
+    end
+  end
+
+  def my_count(*arg)
+    count = 0
+    unless block_given?
+
+      if arg.empty?
+        my_count { |i| i }
+      else
+        my_count { |i| arg[0] == i }
+      end
+    end
+    my_each { |i| count += 1 if yield(i) }
+    count
+  end
```

Comment:
* [ ] my_count returns the number of items in enum through enumeration
* [ ] my_count counts the number of items in enum that are equal to item if an argument is given

Reply:
for Req.1 
```ruby
arr = new Array(100) {rand(0...9)}
puts arr.my_count === arr.count // returns false
```

for Req.2
```ruby
arr = [1,5,8,7,9,6,1,5]
p arr.count(1) === arr.my_count(1) // returns false
```